core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -235,6 +235,23 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
           }
 
           tryBuilder.map(_.result())
+        case o: ConfigObject =>
```

Comment:
@froesef Do the changes above make sense to you? In my clone of your branch they pass all the unit tests except the one which verified the error message. If so would you be willing to incorporate them into your PR? 
I'm not sure if the other authors agree with me, but that's my personal preference.